src/scripts/crafttweaker/craftingUtils.zs

```diff
@@ -10,26 +10,67 @@
 	learning but not for copying and pasting and claiming as your own.
 */
 import crafttweaker.item.IIngredient;
+import crafttweaker.item.IItemStack;
 import crafttweaker.liquid.ILiquidStack;
+import crafttweaker.recipes.ICraftingInfo;
+import crafttweaker.recipes.IRecipeFunction;
 
-function getBucketIngredientFromName(liquidName as string) as IIngredient {
+function getBucketAndTankIngredientFromName(liquidName as string, includeTanks as bool) as IIngredient {
 	var bucketsIngredient as IIngredient = null;
 
 	for bucket in buckets {
 		var bucketIngredient as IIngredient = scripts.crafttweaker.utils.formatBucketIngredient(bucket, liquidName);
 
 		if (!isNull(bucketIngredient)) {
-			if (isNull(bucketsIngredient)) {
-				bucketsIngredient = bucketIngredient;
-			} else {
-				bucketsIngredient |= bucketIngredient;
+			if (includeTanks | !(tanks has bucket)) {
+				if (isNull(bucketsIngredient)) {
+					bucketsIngredient = bucketIngredient;
+				} else {
+					bucketsIngredient |= bucketIngredient;
+				}
 			}
 		}
 	}
 
 	return bucketsIngredient;
 }
 
+function getBucketIngredientFromName(liquidName as string) as IIngredient {
+	return getBucketAndTankIngredientFromName(liquidName, false);
+}
+
+function getBucketAndTankIngredient(liquid as ILiquidStack, includeTanks as bool) as IIngredient {
+	return getBucketAndTankIngredientFromName(liquid.name, includeTanks);
+}
+
 function getBucketIngredient(liquid as ILiquidStack) as IIngredient {
-	return getBucketIngredientFromName(liquid.name);
+	return getBucketAndTankIngredient(liquid, false);
 }
+
+global fluidRecipeFunction as IRecipeFunction = function (output as IItemStack, inputs as IIngredient[string], cInfo as ICraftingInfo) as IItemStack {
```

Comment:
Make static and import craftingUtils where used

Reply:
Shall do